controls/compute.sp

```diff
@@ -0,0 +1,293 @@
+        when d.disk_state = 'Unattached' then d.name || ' not attached to virtual machine.'
+        when m.instance_state = 'running' then d.name || ' attached to running virtual machine.'
+        else d.name || ' not attached to running virtual machine.'
+      end as reason,
+      d.resource_group,
+      sub.display_name as subscription
+    from
+      azure_compute_disk as d
+      left join attached_disk_with_vm as m on (d.name = m.os_disk_name or m.data_disk_names ?| array[d.name])
+      left join azure_subscription as sub on sub.subscription_id = d.subscription_id;
+  EOT
+
+  tags = merge(local.compute_common_tags, {
+    class = "deprecated"
+  })
+}
+
+control "compute_disk_high_iops" {
+  title       = "Disks with high IOPS should be reviewed"
+  description = "High IOPS disks are costly and usage should be reviewed."
+  severity    = "low"
+
+  sql = <<-EOT
+    select
+      disk.id as resource,
+      case
+        when disk_iops_read_write > 16000 then 'alarm'
+        else 'ok'
+      end as status,
+      disk.title || ' has ' || disk_iops_read_write || ' IOPS.'
+      as reason,
+      disk.resource_group,
+      sub.display_name as subscription
+    from
+      azure_compute_disk as disk,
+      azure_subscription as sub
+    where
+      sub.subscription_id = disk.subscription_id;
+  EOT
+
+  tags = merge(local.compute_common_tags, {
+    class = "deprecated"
+  })
+}
+
+control "compute_disk_large" {
+  title       = "Disks with over 100 GB should be resized if too large"
+  description = "Large compute disks are unusual, expensive and should be reviewed."
+  severity    = "low"
+
+  sql = <<-EOT
+    select
+      disk.unique_id as resource,
+      case
+        when disk_size_gb <= 100 then 'ok'
+        else 'alarm'
+      end as status,
+      disk.title || ' is ' || disk_size_gb || ' GB.' as reason,
+      resource_group,
+      sub.display_name as subscription
+    from
+      azure_compute_disk as disk,
+      azure_subscription as sub
+    where
+      sub.subscription_id = disk.subscription_id;
+  EOT
+
+  tags = merge(local.compute_common_tags, {
+    class = "deprecated"
+  })
+}
+
+control "compute_disk_low_iops" {
+  title       = "Disks with low IOPS should be reviewed"
+  description = "Compute disks with low IOPS should be reviewed."
+  severity    = "low"
+
+  sql = <<-EOT
+    select
+      disk.id as resource,
+      case
+        when disk_iops_read_write <= 1500 then 'alarm'
+        else 'ok'
+      end as status,
+      disk.title || ' has ' || disk_iops_read_write || ' IOPS.' as reason,
+      disk.resource_group,
+      sub.display_name as subscription
+    from
+      azure_compute_disk as disk,
+      azure_subscription as sub
+    where
+      sub.subscription_id = disk.subscription_id;
+  EOT
+
+  tags = merge(local.compute_common_tags, {
+    class = "deprecated"
+  })
+}
+
+control "compute_disk_snapshot_storage_standard" {
+  title       = "Disks snapshots storage type should be standard"
```

Comment:
Is this a general recommendation we should put out for all snapshots? Similar to the disk recommendation below, it seems like determining snapshot storage type is an application decision, unless there's a clear cut benefit, like when a new cheaper, better performing generation comes out

Reply:
@cbruno10   This use case is being derived from the section `Use standard snapshots for managed disks` in the below link. Please have a look. 
https://docs.microsoft.com/en-gb/azure/advisor/advisor-cost-recommendations#use-standard-snapshots-for-managed-disks